pygmt/src/triangulate.py

```diff
@@ -0,0 +1,384 @@
+              ``outgrid``)
+        """
+        # Return an xarray.DataArray if ``outgrid`` is not set
+        with GMTTempFile(suffix=".nc") as tmpfile:
+            if isinstance(outgrid, str):
+                output_type = "file"
+            elif outgrid is None:
+                output_type = "xarray"
+                outgrid = tmpfile.name
+            else:
+                raise GMTInvalidInput(
+                    "'outgrid' should be a proper file name or `None`"
+                )
+
+            return triangulate._triangulate(
+                data=data,
+                x=x,
+                y=y,
+                z=z,
+                output_type=output_type,
+                outgrid=outgrid,
+                spacing=spacing,
+                projection=projection,
+                region=region,
+                verbose=verbose,
+                binary=binary,
+                nodata=nodata,
+                find=find,
+                coltypes=coltypes,
+                header=header,
+                incols=incols,
+                registration=registration,
+                skiprows=skiprows,
+                wrap=wrap,
+                **kwargs,
+            )
+
+    @staticmethod
+    @fmt_docstring
+    @kwargs_to_strings(R="sequence")
+    def delaunay_triples(  # pylint: disable=too-many-arguments,too-many-locals
+        data=None,
+        x=None,
+        y=None,
+        z=None,
+        output_type="pandas",
+        outfile=None,
+        projection=None,
+        verbose=None,
+        binary=None,
+        nodata=None,
+        find=None,
+        coltypes=None,
+        header=None,
+        incols=None,
+        skiprows=None,
+        wrap=None,
+        **kwargs,
+    ):
+        """
+        Delaunay triangle based gridding of Cartesian data.
+
+        Reads in x,y[,z] data and performs Delaunay triangulation, i.e., it
+        finds how the points should be connected to give the most equilateral
+        triangulation possible. If a map projection (give *region* and
+        *projection*) is chosen then it is applied before the triangulation is
+        calculated. The actual algorithm used in the triangulations is either
+        that of Watson [1982] or Shewchuk [1996] [Default] (if installed; type
+        **gmt get GMT_TRIANGULATE** on the command line to see which method is
+        selected). This choice is made during the GMT installation.
+        Furthermore, if the Shewchuk algorithm is installed then you can also
+        perform the calculation of Voronoi polygons and optionally grid your
+        data via the natural nearest neighbor algorithm. **Note**: For
+        geographic data with global or very large extent you should consider
+        :gmt-docs:`sphtriangulate <sphtriangulate.html>` instead since
+        **triangulate** is a Cartesian or small-geographic area operator and is
+        unaware of periodic or polar boundary conditions.
```

Comment:
I'm wondering if this section is needed in all of the methods or if it would be sufficient in the class docstring?

If this section stays in this docstring, I recommend removing the parts about Voronoi polygons since I think that would be handled in a separate method.

Reply:
> I'm wondering if this section is needed in all of the methods or if it would be sufficient in the class docstring?

Well, it's not as visible in the class documentation for someone figuring it out from Python/Jupyter. I say we keep it

> If this section stays in this docstring, I recommend removing the parts about Voronoi polygons since I think that would be handled in a separate method.

Ok, I'll remove the Voronoi polygons mention.

Reply:
Sounds good.